velox/exec/tests/RowContainerTest.cpp

```diff
@@ -729,6 +729,23 @@ TEST_F(RowContainerTest, rowSizeWithNormalizedKey) {
   ASSERT_EQ(numRows, 1);
 }
 
+TEST_F(RowContainerTest, estimateRowSize) {
+  auto data = makeRowContainer({BIGINT()}, {VARCHAR()});
+  auto allRows = 100'000;
+  for (auto i = 0; i < allRows; i++) {
+    data->newRow();
+  }
+
+  RowContainerIterator iter;
+  char* rows[allRows];
+  // Test sizes smaller than the total row container size.
+  std::vector<int64_t> sizes = {1, 1 << 10, 1 << 20};
+  for (auto size : sizes) {
```

Comment:
Would you explain how this test works? I'm surprised to see a call to listRows as I expect estimateRowSize to not be affected by a call to listRows. However, I'd expect estimateRowSize to depend on how much variable-size data has been written. It would be nice to write not only fixed-width rows, but also variable-width rows.

Reply:
Sorry, it's a bit confusing. I renamed the variables, hopefully, to make it more readable.

listRows() doesn't have to do with estimateRowSize(). listRows() is used to get the fact or the expected numRows given the different batchSizes. Then EXPECT_EQ says the data->estimateRowSize() is accurate to derive the actual num of rows that matches the expected numRows.